notebooks/bigQuery.ipynb

```diff
@@ -0,0 +1,214 @@
+    "DECLARE end_date STRING DEFAULT \"20211101\";\n",
+    "\n",
+    "WITH sessions AS (\n",
+    "SELECT DISTINCT\n",
+    "    CONCAT(fullVisitorId, \"-\", CAST(visitId AS STRING)) AS sessionId,\n",
+    "    REGEXP_REPLACE(hits.page.pagePath, r'[?#].*', '') AS pagePath\n",
+    "FROM `govuk-bigquery-analytics.87773428.ga_sessions_*`\n",
+    "CROSS JOIN UNNEST(hits) AS hits\n",
+    "WHERE\n",
+    "    _TABLE_SUFFIX BETWEEN start_date AND end_date\n",
+    "    AND hits.page.pagePath NOT LIKE \"/print%\"\n",
+    "    AND hits.type = 'PAGE'\n",
+    "    AND (SELECT value FROM hits.customDimensions WHERE index = 2) NOT IN {} \n",
+    ")\n",
+    "\n",
+    "SELECT \n",
+    "    pagePath,\n",
+    "    COUNT(sessionId) AS noOfSessions \n",
+    "FROM sessions\n",
+    "GROUP BY pagePath\n",
+    "HAVING noOfSessions >= 5\n",
+    "ORDER BY noOfSessions\n",
+    "\"\"\".format(\n",
+    "    ignore\n",
+    ")\n",
+    "\n",
+    "# create Data Frame\n",
+    "page_paths = pandas_gbq.read_gbq(sql, project_id=project_id)\n",
+    "page_paths.to_csv(\"page_paths.csv\", index=False)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7832e905",
+   "metadata": {},
+   "source": [
+    "### By page hits per content ID"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "5d602041",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# get content ids with more than 5 page hits in the last 2 weeks,\n",
+    "# ignoring irrelevant pages\n",
+    "sql = \"\"\"\n",
+    "DECLARE start_date STRING DEFAULT \"20211018\";\n",
+    "DECLARE end_date STRING DEFAULT \"20211101\";\n",
+    "\n",
+    "WITH sessions AS (\n",
+    "SELECT DISTINCT\n",
+    "    CONCAT(fullVisitorId, \"-\", CAST(visitId AS STRING)) AS sessionId,\n",
+    "    (SELECT value FROM hits.customDimensions WHERE index = 4) AS contentId\n",
+    "FROM `govuk-bigquery-analytics.87773428.ga_sessions_*`\n",
+    "CROSS JOIN UNNEST(hits) AS hits\n",
+    "WHERE\n",
+    "    _TABLE_SUFFIX BETWEEN start_date AND end_date\n",
+    "    AND hits.page.pagePath NOT LIKE \"/print%\"\n",
+    "    AND hits.type = 'PAGE'\n",
+    "    AND (SELECT value FROM hits.customDimensions WHERE index = 2) NOT IN {}\n",
+    ")\n",
+    "\n",
+    "SELECT \n",
+    "    contentID,\n",
+    "    COUNT(sessionId) AS noOfSessions \n",
+    "FROM sessions\n",
+    "GROUP BY contentID\n",
+    "HAVING noOfSessions >= 5\n",
+    "\"\"\".format(\n",
+    "    ignore\n",
+    ")\n",
+    "\n",
+    "# create Data Frame\n",
+    "content_ids = pandas_gbq.read_gbq(sql, project_id=project_id)\n",
+    "content_ids.to_csv(\"content_ids.csv\", index=False)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6f803373",
+   "metadata": {},
+   "source": [
+    "### Get SaB pages"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "1d48895a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# get all the SaB pages\n",
```

Comment:
As querying the sabpages from BigQuery costs money, and they originally live in a csv file, I think it might be best to direct users to access the csv from google drive. In the documentation we can put 'File sab_pages.csv can be found in DataLabs shared GDrive in folder GOV.UK teams/2021-2022/Data labs/14 Network Analysis'. I spoke to Alessia about this and she agreed. This also requires authentication, so covers our back there.  So, we could just remove this query. Then in the neo4j to networkx bigquery notebook where you have: 

```
# create Data Frame 
sab_pages = pd.read_csv("sab_pages.csv")
```
 
You can change this to:
 
```
# Import libraries 
import pandas as pd 
import gspread 
from oauth2client.service_account import ServiceAccountCredentials
 
# Connect to service account
scope = ['https://spreadsheets.google.com/feeds'] 
credentials = ServiceAccountCredentials.from_json_keyfile_name(GOOGLE_APPLICATION_CREDENTIALS, scope) 
gc = gspread.authorize(credentials)
 
# Import the data from google sheets into Jupyter
spreadsheet_key = '1x3lHpUkIm-KTnWrRJU11An1J2Yb4KJT8z0QpuXOm3e0' 
book = gc.open_by_key(spreadsheet_key) 
worksheet = book.worksheet("sab_pages") 
table = worksheet.get_all_values()
 
# Convert table data into a dataframe 
pd.DataFrame(table[1:], columns=table[0])
```

Note how this also uses the secret credential `GOOGLE_APPLICATION_CREDENTIALS`, so we need to make sure we have (but with your path to the file)
`export GOOGLE_APPLICATION_CREDENTIALS="/Users/alicestephenson/gds/secrets/govuk-bigquery-analytics-credentials.json` in the .secrets file. 
 
We also need to have 
`GOOGLE_APPLICATION_CREDENTIALS = os.getenv("GOOGLE_APPLICATION_CREDENTIALS")` in the script.